### ISSUE TITLE
[BUG] BlobClient.GenerateSasUri(BlobSasPermissions, DateTimeOffset) ignores timezone for expiration time

### ISSUE THREAD

**Issue by ChristianHehlBM** [Azure/azure-sdk-for-net#23440]
**Describe the bug**
BlobClient.GenerateSasUri(BlobSasPermissions.Read, DateTimeOffset.Now) returns a URI with a different expiration time than 
BlobClient.GenerateSasUri(BlobSasPermissions.Read, DateTimeOffset.Now.ToUniversalTime()) when in a non UTC timezone. 

**Expected behavior**
The expiration time should be the same since the point in time of both values is the same.

**Actual behavior (include Exception or Stack Trace)**
It returns a different expiration time (offset by the timezone of the machine)

**To Reproduce**

Consider the following test:

```.cs
[Test]
public void SameExpirationTime()
{
    var now = DateTimeOffset.Now; // the machine is in GMT+2 right now
    var utcNow = now.ToUniversalTime();
    var blobClient = new BlobClient("yourConnectionString","yourBlobContainerName", "yourBlobName");
    var uriNow = blobClient.GenerateSasUri(BlobSasPermissions.Read, now);
    var uriUtcNow = blobClient.GenerateSasUri(BlobSasPermissions.Read, utcNow);

    var expiresOnNow = HttpUtility.ParseQueryString(uriNow.Query).Get("se");
    var expiresOnUtcNow = HttpUtility.ParseQueryString(uriUtcNow.Query).Get("se");
    
    Assert.AreEqual(DateTimeOffset.Parse(expiresOnNow), DateTimeOffset.Parse(expiresOnUtcNow));
}
```

The result is:

```
Expected: 2021-08-20 10:14:11+00:00
But was:  2021-08-20 08:14:11+00:00
```

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.9.1
 - Hosting platform or OS: macOS
 - .NET version: 5.0.301
 - IDE and version : JetBrains Rider V2021.1.3

**Comment by amnguye** [comment]
There's a problem with the value you're passing. You're passing a `DateTime` object now instead of a `DateTimeOffset` object. So when your code goes to convert from `DateTime` to `DateTimeOffset`, it will do the UTC offset based on the local time offset.

Please see the documentation about `DateTime.ToUniversalTime`.
https://docs.microsoft.com/en-us/dotnet/api/system.datetime.touniversaltime?view=net-5.0#remarks

In order to avoid the conversion and have the correct offset in UTC time please use `DateTImeOffset.UtcNow` instead. This will achieve what you are looking without a dependency on the timezone set in different environments.
https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.utcnow?view=net-5.0

**Comment by ChristianHehlBM** [comment]
I don't see how i am passing in a DateTime object instead of a `DateTimeOffset`. I Use `DateTimeOffset.Now` which gives me a DateTimeOffset object, and then use `DateTimeOffset.ToUniversalTime()` which also returns a `DateTimeOffset` object. In fact when checking the ticks and utcTicks properties of both object **they refer to exactly the same point in time.**

I also compared `DateTimeOffset.Now.ToUniversalTime()` and `DateTimeOffset.UtcNow` and they both result in the exactly same expiration time. The issue is just with DateTimeOffset Values where the Offset is not 0.

To summarize:

`DateTimeOffset.Now`, `DateTimeOffset.Now.ToUniversalTime()` and `DateTimeOffset.UtcNow` all refer to the same point in time (Ticks and UtcTicks Property identical), but `DateTimeOffset.Now` results in a different expiration time than the other two.

In my opinion handing in the same point in time should result in the same expiration time, no matter in which timezone the value was created.

**Comment by amnguye** [comment]
Yes I agree the issue is with the offset. Because you're handing a DateTimeOffset that is not 0, you're getting an expirey time with this offset.

The SDK cannot alter the offset passed by the user, it will start giving users who are taking a dependency on that offset to create their expirey time, a different expirey time. This will break customers dependent on the offset to be taken in consideration of the expirey time. (breaking change)

The user should be able to handle and detect what time zone their application is running and set the expirey time based on that. The recommended approach to not be affected by the offset property is to use DateTimeOffset.UtcNow.
